inc/msp/msp_msg.hpp

```diff
@@ -4,16 +4,28 @@
 #ifndef MSP_MSG_HPP
 #define MSP_MSG_HPP
 
+#include "Message.hpp"
+
 #include <string>
 #include <array>
 #include <sstream>
 #include <set>
 #include <climits>
 #include <cassert>
+#include <iomanip>
+#include <vector>
+
+/*================================================================
+ * actual messages have id and the relevant encode decode methods
+ * the logic for encoding and decoding must be within a message-derived class
+ * non message-derived structs must have pack/unpack subroutines
+ * 
+ */
 
-#include "types.hpp"
+//because the gnu c std lib is stupid... 
+#undef major
+#undef minor
```

Comment:
`major` and `minor` are used as fields in `ApiVersion`. But as the comment says, GNU C Std Lib is stupid and `#define`s those as other things.

Reply:
I see. That's unfortunate :-)
Maybe we could also rename our variables to `api_major` and `api_minor`?

It's not nice that they did not use namespaces for their macros but I would like to prevent such `undef`s. This issue is actually known (https://sourceware.org/ml/libc-alpha/2017-02/msg00079.html): 
> The inclusion of <sys/sysmacros.h> by <sys/types.h> is deprecated.
This means that in a future release, the macros “major”, “minor”, and
“makedev” will only be available from <sys/sysmacros.h>.

We are not including `sys/sysmacros.h` in our code. I am wondering if this comes from asio somehow.

If there is no way around this (e.g. renaming our variables), then at least place a comment there why these `undef`s are required there

Reply:
its because the gnu c std lib is stupid :-)